src/gxcore/common/geolocation.ts

```diff
@@ -0,0 +1,241 @@
+import { notImplemented } from "../../misc/helpers";
+
+export class GeneXusCommonGeolocation {
+
+  /**
+   * Returns the current location for the device
+   * @param minAccuracy
+   * @param timeout
+   * @param includeHeadingAndSpeed
+   * @param ignoreErrors
+   * @return any
+   */
+  public static getMyLocation(minAccuracy: number, timeout: number, includeHeadingAndSpeed:boolean, ignoreErrors:boolean): Promise<GeneXusCommonGeolocationInfoData> {
+    return new Promise<GeneXusCommonGeolocationInfoData>((resolve, reject) => {
+      if (navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition((position: any) => { 
+          let geoPosition = new GeneXusCommonGeolocationInfoData();
+          geoPosition.Location = position.coords.latitude.toString() + "," + position.coords.longitude.toString();
+          geoPosition.Heading = position.coords.heading;
+          geoPosition.Precision = position.coords.accuracy;
+          geoPosition.Description = "";
+          geoPosition.Speed = position.coords.speed;
+          geoPosition.Time = position.timestamp;
+          resolve(geoPosition);
+        }, 
+        (error) => {
+          if (ignoreErrors) { 
+            let geoPosition = new GeneXusCommonGeolocationInfoData();
+            geoPosition.Location = "";
+            geoPosition.Heading = 0;
+            geoPosition.Precision = 0;
+            geoPosition.Description = "";
+            geoPosition.Speed = 0;
+            geoPosition.Time = null;
+              resolve( geoPosition);
+          } else {
+            reject( "Could not get location information ("+error.message+")");
+          }
+        },
+        {timeout: timeout/1000});
```

Comment:
`timeout` parameter, [as defined in GeneXus](https://wiki.genexus.com/commwiki/servlet/wiki?25164,GetMyLocation+method), is expressed in seconds. The function [`navigator.geolocation.getCurrentPosition`](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/getCurrentPosition) takes a timeout value in milliseconds.

I think the conversion is wrong and should be the other way around...
```suggestion
        {timeout: timeout * 1000});
```